x-pack/plugins/security_solution/public/common/hooks/endpoint/use_alert_response_actions_support.ts

```diff
@@ -0,0 +1,241 @@
+  const isAlert = useMemo(() => {
+    return some({ category: 'kibana', field: 'kibana.alert.rule.uuid' }, eventData);
+  }, [eventData]);
+
+  const agentType: ResponseActionAgentType | undefined = useMemo(() => {
+    if ((find({ field: 'agent.type' }, eventData)?.values ?? []).includes('endpoint')) {
+      return 'endpoint';
+    }
+
+    const eventModuleValues = find({ field: 'event.module' }, eventData)?.values ?? [];
+
+    if (eventModuleValues.includes('sentinel_one')) {
+      return 'sentinel_one';
+    }
+
+    if (eventModuleValues.includes('crowdstrike')) {
+      return 'crowdstrike';
+    }
+
+    return undefined;
+  }, [eventData]);
+
+  const isFeatureEnabled: boolean = useMemo(() => {
+    return agentType ? isAgentTypeAndActionSupported(agentType) : false;
+  }, [agentType]);
+
+  const agentId: string = useMemo(() => {
+    if (!agentType) {
+      return '';
+    }
+
+    if (agentType === 'endpoint') {
+      return getAlertDetailsFieldValue({ category: 'agent', field: 'agent.id' }, eventData);
+    }
+
+    if (agentType === 'sentinel_one') {
+      return getAlertDetailsFieldValue(
+        { category: 'observer', field: RESPONSE_ACTIONS_ALERT_AGENT_ID_FIELD.sentinel_one },
+        eventData
+      );
+    }
+
+    if (agentType === 'crowdstrike') {
+      return getAlertDetailsFieldValue(
+        { category: 'crowdstrike', field: RESPONSE_ACTIONS_ALERT_AGENT_ID_FIELD.crowdstrike },
+        eventData
+      );
+    }
+
+    return '';
+  }, [agentType, eventData]);
+
+  const doesHostSupportResponseActions = useMemo(() => {
+    return Boolean(isFeatureEnabled && isAlert && agentId && agentType);
+  }, [agentId, agentType, isAlert, isFeatureEnabled]);
+
+  const agentIdField = useMemo(() => {
+    if (agentType) {
+      return RESPONSE_ACTIONS_ALERT_AGENT_ID_FIELD[agentType];
+    }
+
+    return '';
+  }, [agentType]);
+
+  const supportedActions = useMemo(() => {
+    return RESPONSE_ACTION_API_COMMANDS_NAMES.reduce<AlertAgentActionsSupported>(
+      (acc, responseActionName) => {
+        acc[responseActionName] = false;
+
+        if (agentType && isFeatureEnabled) {
+          acc[responseActionName] = isAgentTypeAndActionSupported(
+            agentType,
+            responseActionName,
+            'manual'
+          );
+        }
+
+        return acc;
+      },
+      {} as AlertAgentActionsSupported
+    );
+  }, [agentType, isFeatureEnabled]);
+
+  const hostName = useMemo(() => {
+    // TODO:PT need to check if crowdstrike event has `host.name`
+    if (agentType === 'crowdstrike') {
+      return getAlertDetailsFieldValue(
+        { category: 'crowdstrike', field: 'crowdstrike.event.HostName' },
+        eventData
+      );
+    }
+
+    return getAlertDetailsFieldValue({ category: 'host', field: 'host.name' }, eventData);
+  }, [agentType, eventData]);
+
+  const platform = useMemo(() => {
+    // TODO:PT need to check if crowdstrike event has `host.os.family`
```

Comment:
I tested against 2 different sources - falcon and fdr data_streams. It's sometimes tricky to understand those because these 2 were implemented separately and some mappings are just different ;/ 

However, I think we can assume that it contains one of these:
- host.os.type
- host.os.platform

Reply:
I can adjust these in the follow up PR 👍